### PR TITLE
README: improve readability and use standard layout

### PR DESCRIPTION
**Pull request by zimbatm**
Improve the landing page.

The goal is that people can read the first 2 paragraph and quickly figure out if the project is for them or not. Then how to use it. And finally get into the more hacky things.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
 
-All it does is copy the release tarballs from Hydra and tweak the install
-script so that it fetches them from GitHub instead.
+It's mainly useful if you want to test changes in the Nix project before it
+becomes available in an official release. Or if you like living on the edge.
 
 ## Usage
 
-Pick the release that you want to use. Each release has a usage instruction attached to it: <https://github.com/numtide/nix-unstable-installer/releases/latest>
+Pick the release that you want to use. Each release has a usage instruction
```
Comment (lilyinstarlight) [inline README.md:13]:
```suggestion
Pick the release that you want to use. Each release has usage instructions
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+* publish all of that as a github release.
+
+## Contributing
+
+Contributions and discussions are welcome. Please make sure to send a WIP PR
+or issue before doing large refactors, so your work doesn't get wasted (in
+case of disagreement).
+
+A big thanks to [@lilyinstarlight](https://github.com/lilyinstarlight) who
+took this project from its hacky state to a something well finished.
```
Comment (lilyinstarlight) [inline README.md:35]:
```suggestion
took this project from its hacky state to something well finished.
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,40 @@
 # Nix Unstable Installer
 
-This project is about making Nix unstable releases available to the general
-public. It allows to test and use preview features like Flakes and provide
-early feedback.
+This project is a companion project to
+[nixos/nix](https://github.com/nixos/nix). It takes the latest successful
```
Comment (lilyinstarlight) [inline README.md:4]:
```suggestion
[NixOS/nix](https://github.com/NixOS/nix). It takes the latest successful
```

**Review by lilyinstarlight — APPROVED** [review]
This looks fabulous! I think it will be much better and more instructive for people finding this repo now :)

I have a few minor grammar/capitalization nitpicks, but I think this is a really good update

**Comment by zimbatm** [comment]
thanks for the review!

**Pull request merged**